Framework/Algorithms/inc/MantidAlgorithms/SpectrumAlgorithm.h

```diff
@@ -88,7 +88,7 @@ class MANTID_ALGORITHMS_DLL SpectrumAlgorithm : public API::Algorithm {
     auto size = static_cast<int64_t>(indexSet.size());
     API::Progress progress(this, 0.0, 1.0, size);
 
-    PARALLEL_FOR1((&workspace))
+    PARALLEL_FOR_IF(Kernel::threadSafe(workspace))
```

Comment:
I like this. It's nice that it now conveys that there is actually a `if` going on and it also describes the check too.